services/121-service/src/transaction-queues/dto/safaricom-transaction-job.dto.ts

```diff
@@ -0,0 +1,47 @@
+import {
+  IsBoolean,
+  IsNotEmpty,
+  IsNumber,
+  IsNumberString,
+  IsOptional,
+  IsString,
+} from 'class-validator';
+
+export class SafaricomTransactionJobDto {
+  @IsNotEmpty()
+  @IsNumberString()
+  public readonly programId: number;
+
+  @IsNotEmpty()
+  @IsNumberString()
+  public readonly paymentNumber: number;
+
+  @IsNotEmpty()
+  @IsString()
+  public readonly referenceId: string;
+
+  @IsNotEmpty()
+  @IsNumber()
+  public readonly transactionAmount: number;
+
+  @IsNotEmpty()
+  @IsBoolean()
+  public readonly isRetry: boolean;
+
+  @IsNotEmpty()
+  @IsNumber()
+  public readonly userId: number;
+
+  @IsNotEmpty()
+  @IsNumber()
+  public readonly bulkSize: number;
+
+  @IsOptional()
+  public readonly phoneNumber?: string;
+
+  @IsOptional()
+  public readonly idNumber?: string;
+
+  @IsOptional()
```

Comment:
Why optional? Only if not a retry?

And I suggest to put this attribute higher: aim to put "more important" higher and "details" lower.

Reply:
We copied this setup from Visa example. Let's discuss.